apps/src/music/blockly/blocks/simple2.js

```diff
@@ -159,6 +164,36 @@ export const playSoundAtCurrentLocationSimple2 = {
     `
 };
 
+export const playPatternAtCurrentLocationSimple2 = {
+  definition: {
+    type: BlockTypes.PLAY_PATTERN_AT_CURRENT_LOCATION_SIMPLE2,
+    message0: 'play pattern %1',
+    args0: [fieldPatternDefinition],
+
+    inputsInline: true,
+    previousStatement: null,
+    nextStatement: null,
+    style: 'music_blocks',
+    tooltip: 'play pattern',
+    helpUrl: ''
+  },
+  generator: block =>
+    `
+      MusicPlayer.playPatternAtMeasureById(
+        ${JSON.stringify(block.getFieldValue(FIELD_PATTERN_NAME))},
+        ProgramSequencer.getCurrentMeasure(),
+        __insideWhenRun,
+        null,
+        __currentFunction,
+        RandomSkipManager.getSkipContext(),
+        __effects
+      );
+      ProgramSequencer.updateMeasureForPlayByLength(
+        1
```

Comment:
Minor, but can we save this as a constant, i.e. DEFAULT_PATTERN_LENGTH? It can be reused in MusicPlayer#getLengthForId, and would be easier to spot if/when we allow multi-measure patterns

Reply:
Good call.